<span style="color:black">Question  1A:

In [6]:
import pandas as pd
from scipy.stats import kurtosis, skew

df = pd.read_csv("problem1.csv")
data = df["X"]
mean = data.mean()
variance = data.var()
skewness = skew(data)
kurt = kurtosis(data)
print(f"Mean: {mean}")
print(f"Variance: {variance}")
print(f"Skewness: {skewness}")
print(f"Kurtosis: {kurt}")

Mean: 0.05019795790476916
Variance: 0.010332476407479588
Skewness: 0.1204447119194402
Kurtosis: 0.2229270674503816


<span style="color:black">Question 1B:

<span style="font-size:15px;color:black">It depends on whether the distribution has a big population. If the sample size is bigger than 30, I will choose normal distribution, and if the sample size is smaller than 30, I will choose T-distribution. It also depends on whether I know the population standard deviation. If it is known, there is high chance that I choose normal distribution. Still, if the data is known to be skewed or has extreme outliers, I would choose T-distribution over normal distribution since a dataset with more symmetricity percularities is more fit for normal distribution. Normal distribution would be forced to be used somehow under the condition that normality is assumed like linear regression. Under this situation, there is a slight skewness, so I decide to choose T-distribution. 

Question 1C:

In [9]:
from scipy.stats import norm
from scipy.stats import t
from scipy.stats import kstest


df_t, loc, scale = t.fit(data)
print(f"T Distribution Fit: df = {df_t}, loc = {loc}, scale = {scale}")
mu, sigma = norm.fit(data)
print(f"Normal Distribution Fit: Mean = {mu}, Std = {sigma}")

ks_normal = kstest(data, 'norm', args=(mu, sigma))
ks_t = kstest(data, 't', args=(df_t, loc, scale))
print(f"Normal K-S Test: {ks_normal}")
print(f"T Distribution K-S Test: {ks_t}")


T Distribution Fit: df = 28.71016692613074, loc = 0.04986675416872735, scale = 0.09800128766123102
Normal Distribution Fit: Mean = 0.05019795790476916, Std = 0.10159795239606018
Normal K-S Test: KstestResult(statistic=np.float64(0.01275165971875869), pvalue=np.float64(0.9962422791034079), statistic_location=np.float64(0.1931202657983203), statistic_sign=np.int8(1))
T Distribution K-S Test: KstestResult(statistic=np.float64(0.012697894723120384), pvalue=np.float64(0.996453485092733), statistic_location=np.float64(-0.0377745593740776), statistic_sign=np.int8(1))


<span style="font-size:15px;color:black">Based on the test result provided above, both T-distribution and normal distribution has high p-values with 0.99624 for Normal distribution and 0.99645 for T-distribution. For practical application, both could be implemented with confidency. T-distribution has a slightly lower statistic data then normal distribution, so T-distribution is actually a little bit better than normal distribution under this dataset. Therefore, my choice matches with data. 